proxy/InkAPI.cc

```diff
@@ -9112,6 +9113,66 @@ TSSslContextDestroy(TSSslContext ctx)
   SSLReleaseContext(reinterpret_cast<SSL_CTX *>(ctx));
 }
 
+void
+TSRegisterProtocolSet(TSVConn sslp, TSNextProtocolSet ps)
+{
+  NetVConnection *vc        = reinterpret_cast<NetVConnection *>(sslp);
+  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(vc);
+  if (ssl_vc) {
+    ssl_vc->clearnpnSet();
```

Comment:
I discussed this with @persiaAziz and we concluded the `assert` is no longer useful given that the purpose of this API is to change the value even if it's already set. Therefore these two methods should be unified and the `assert` removed.